lib/matplotlib/animation.py

```diff
@@ -619,12 +650,22 @@ def _init_from_registry(cls):
                 binpath = ''
         rcParams[cls.exec_key] = rcParamsDefault[cls.exec_key] = binpath
 
+    @classmethod
+    def isAvailable(cls):
+        '''
+        Check to see if a MovieWriter subclass is actually available by
+        running the commandline tool.
+        '''
+        bin_path = cls.bin_path()
+        if bin_path == "convert":
+            cls._init_from_registry()
+        return super(ImageMagickBase, cls).isAvailable()
 
 ImageMagickBase._init_from_registry()
 
 
 @writers.register('imagemagick')
-class ImageMagickWriter(MovieWriter, ImageMagickBase):
+class ImageMagickWriter(ImageMagickBase, MovieWriter, ):
```

Comment:
Hm, I did this to get `ImageMagickWriter.isAvailable()` picked up from `ImageMagickBase` and not from `MovieWriter` (both declare this class method, so one of them is first and this ordering made the `ImageMagickBase` the first, which then calls the same method in the other class via super). 

-> try and error...